### PR TITLE
Builder annotation support

### LINKED ISSUE #5106: Add possibility to configure annotation for the builder pods
### Requirement

As a user of camel-k, I would like to be able to specify custom annotations that would be applied to the builder pods. 

### Problem

In the context of a Kubernetes cluster that uses AppArmor, the buildah strategy uses containers that are restricted. 
In order to lift that restriction, we must unconfine the buildah container. This can be done using the following annotation which must be present on the builder pod 

```
container.apparmor.security.beta.kubernetes.io/buildah: unconfined 
```

### Proposal

Extend the current buildConfiguration to include custom annotations.

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by rodcloutier**
This PR proposes to add the possibility to configure annotation on the build pod. 

This was required to allow build pods to properly execute in the context of using Buildah in an AppArmor enabled Kubernetes cluster. 

Also included are a few tweak encountered while running the operator out of cluster. 

**Release Note**
```release-note
Add possibility to use custom annotations on builder pods.
```

Fixes #5106

### CODE REVIEW

**Comment by rodcloutier** [comment]
I am struggling to find any place to insert test for this enhancement. Could anyone chime in?

File: pkg/apis/camel/v1/common_types.go [hunk 1]
```diff
@@ -55,6 +55,8 @@ type BuildConfiguration struct {
 	LimitMemory string `property:"limit-memory" json:"limitMemory,omitempty"`
 	// The node selector for the builder pod. Only used for `pod` strategy
 	NodeSelector map[string]string `property:"node-selector" json:"nodeSelector,omitempty"`
+	// Annotation to use for the builder pod. Only user for `pod` strategy
```
Comment (squakez) [inline pkg/apis/camel/v1/common_types.go:58]:
```suggestion
	// Annotation to use for the builder pod. Only used for `pod` strategy
```

File: pkg/cmd/operator/operator.go [hunk 2]
```diff
@@ -165,8 +165,8 @@ func Run(healthPort, monitoringPort int32, leaderElection bool, leaderElectionID
 		// in which case it's not possible to determine a namespace.
 		operatorNamespace = watchNamespace
 		if operatorNamespace == "" {
-			leaderElection = false
-			log.Info("unable to determine namespace for leader election")
+			// We cannot go forward anyhow since later on we need a namespace to create resources
```
Comment (squakez) [inline pkg/cmd/operator/operator.go:168]: Please, open a separate PR for this change so we can have a more focused look on what it really involves.

File: pkg/cmd/operator/operator.go [hunk 3]
```diff
@@ -295,7 +295,8 @@ func getOperatorImage(ctx context.Context, c ctrl.Reader) (string, error) {
 	ns := platform.GetOperatorNamespace()
 	name := platform.GetOperatorPodName()
 	if ns == "" || name == "" {
-		return "", nil
+		// We are most likely running out of cluster. Let's take a chance and use the default value
```
Comment (squakez) [inline pkg/cmd/operator/operator.go:298]: Ditto

File: pkg/controller/integrationkit/build.go [hunk 4]
```diff
@@ -133,8 +133,10 @@ func (action *buildAction) handleBuildSubmitted(ctx context.Context, kit *v1.Int
 			}
 		}
 		// The build operation, when executed as a Pod, should be executed by a container image containing the
-		// `kamel builder` command. Likely the same image running the operator should be fine.
-		buildConfig.ToolImage = platform.OperatorImage
+		// `kamel builder` command. If not specified, likely the same image running the operator should be fine.
```
Comment (squakez) [inline pkg/controller/integrationkit/build.go:136]: I think we must force the image to be the one on which the operator is running. Please, open a separate PR to discuss this separately.

**Review by squakez — CHANGES_REQUESTED** [review]
Please, open separate PRs in order to discuss each different topic separately. Regarding this specific feature, LGTM. For testing, please, have a look at this example: https://github.com/apache/camel-k/blob/1db0f579416bec989ca6622720f931d2958e64e1/pkg/trait/builder_test.go#L466

File: pkg/cmd/operator/operator.go (hunk 2 above)
Comment (rodcloutier) [reply pkg/cmd/operator/operator.go]: Done see #5109

File: pkg/cmd/operator/operator.go (hunk 3 above)
Comment (rodcloutier) [reply pkg/cmd/operator/operator.go]: Done see #5109. 
I have taken the liberty of bundling these two changes as:
* They are scope to the same issue 
* They are located in the same context/file
* They are small enough

File: pkg/apis/camel/v1/common_types.go (hunk 1 above)
Comment (rodcloutier) [reply pkg/apis/camel/v1/common_types.go]: Updated with the new commits (changing this triggers many other changes when generating files)

File: pkg/controller/integrationkit/build.go (hunk 4 above)
Comment (rodcloutier) [reply pkg/controller/integrationkit/build.go]: Done in #5110

**Review by squakez — APPROVED** [review]
+1 for the controller test. That's the way to improve quality! Thanks!

**Review by gansheer — APPROVED** [review]

**Pull request merged**